scheduler/server/job_state.go

```diff
@@ -264,3 +270,38 @@ func (j *jobState) removeTaskFromStartTimeMap(jobID string, taskID string, start
 		delete(j.tasksByJobClassAndStartTimeSec, timeBucket)
 	}
 }
+
+func (j *jobState) logInconsistentStateValues() {
+	if j.TasksCompleted != len(j.Completed) {
+		log.Errorf("job:%s,%s,%s: TasksCompleted count (%d) != number entries in Completed map (%d)",
+			j.jobClass, j.Job.Def.Requestor, j.Job.Id, j.TasksCompleted, len(j.Completed))
+	}
+	if j.TasksRunning != len(j.Running) {
+		log.Errorf("job:%s,%s,%s: TasksRunning count (%d) != number entries in Running map (%d)",
+			j.jobClass, j.Job.Def.Requestor, j.Job.Id, j.TasksRunning, len(j.Running))
+	}
+	notStarted := len(j.Tasks) - j.TasksRunning - j.TasksCompleted
+	if notStarted != len(j.NotStarted) {
+		log.Errorf("job:%s,%s,%s: TasksRunning count (%d) != number entries in Running map (%d)",
+			j.jobClass, j.Job.Def.Requestor, j.Job.Id, notStarted, len(j.NotStarted))
+	}
+	// TODO remove is slows things down too much
```

Comment:
typo: "remove if this slows..."
should we also add that we don't intend to release this past staging?